sdk/metric/controller/push/push.go

```diff
@@ -153,14 +156,36 @@ func (c *Controller) Stop() {
 	c.ch = nil
 	c.wg.Wait()
 	c.ticker.Stop()
+	c.ticker = nil
 
 	c.tick()
+
+	close(c.configLoaderCh)
+}
+
+func (c *Controller) RestartTicker(period time.Duration) {
```

Comment:
This pretty much becomes the implementation of the config watcher's OnUpdatedConfig method.